tests/test_aws.py

```diff
@@ -76,17 +78,74 @@ def test_build_aws_params():
 ./config.sh --url https://github.com/omsf-eco-infra/awsinfratesting --token test --labels label --ephemeral
 ./run.sh
 """,
+        "TagSpecifications": [
+            {
+                "ResourceType": "instance",
+                "Tags": [
+                    {"Key": "Name", "Value": "test"},
+                    {"Key": "Owner", "Value": "test"},
+                ],
+            }
+        ],
     }
 
 
+def test_create_instance_with_labels(aws):
+    aws.labels = "test"
+    ids = aws.create_instances()
+    assert len(ids) == 1
+
+
 def test_create_instances(aws):
     ids = aws.create_instances()
     assert len(ids) == 1
 
 
+def test_create_instances_missing_release(aws):
+    aws.runner_release = ""
+    with pytest.raises(
+        ValueError, match="No runner release provided, cannot create instances."
+    ):
+        aws.create_instances()
+
+
+def test_create_instances_missing_home_dir(aws):
+    aws.home_dir = ""
+    with pytest.raises(
+        ValueError, match="No home directory provided, cannot create instances."
+    ):
+        aws.create_instances()
+
+
+def test_create_instances_missing_tokens(aws):
+    aws.gh_runner_tokens = []
+    with pytest.raises(
+        ValueError,
+        match="No GitHub runner tokens provided, cannot create instances.",
+    ):
+        aws.create_instances()
+
+
+def test_create_instances_missing_image_id(aws):
+    aws.image_id = ""
+    with pytest.raises(
+        ValueError, match="No image ID provided, cannot create instances."
+    ):
+        aws.create_instances()
+
+
+def test_create_instances_missing_instance_type(aws):
+    aws.instance_type = ""
+    with pytest.raises(
+        ValueError, match="No instance type provided, cannot create instances."
+    ):
+        aws.create_instances()
+
+
 def test_instance_running(aws):
     ids = aws.create_instances()
     assert len(ids) == 1
+    ids = list(ids.keys())
```

Comment:
option to simplify here (and in a few other places this repeats):

`list(ids)` will be the same as `list(ids.keys())` (assuming `ids` is a dict/other mapping that iterates over keys)

Not necessarily a big deal; I sometimes use `list(dct.keys())` to put in the reminder that this object is a dict, but the more "recommended" style is to omit `.keys()`.

Reply:
Fixed in [e96012d](https://github.com/omsf-eco-infra/gha-runner/pull/8/commits/e96012df039841f78d628955c1ef43fca840af5b)